src/main/java/model/Wallet.java

```diff
@@ -0,0 +1,60 @@
+package model;
+
+import exception.InvalidMoneyAmountException;
+
+public class Wallet {
+    private int money;
+
+    /**
+     * @param money: integer
+     * @pre money >= 0
+     */
+    public Wallet(int money) {
+        this.money = Math.max(money, 0);
```

Comment:
To me it is not clear why setting a negative number here results in a balance of 0. But with the method put, an exception is thrown.